providers/implementations/digests/sha3_prov.c

```diff
@@ -194,6 +194,23 @@ static int s390x_keccak_final(unsigned char *md, void *vctx) {
     return 1;
 }
 
+static int s390x_kmac_final(unsigned char *md, void *vctx) {
+    KECCAK1600_CTX *ctx = vctx;
+    size_t bsz = ctx->block_size;
+    size_t num = ctx->bufsz;
+
+    if (!ossl_prov_is_running())
+        return 0;
+    if (ctx->md_size == 0)
+        return 1;
+    memset(ctx->buf + num, 0, bsz - num);
+    ctx->buf[num] = 0x04;
```

Comment:
I note this function only differs by this parameter. It might be more clear to factor the two into one and have the function called by two stub functions used to get function pointers. Up to you though; approving.

Reply:
Good idea.  Done.  Thanks for the suggestion.